NodeJS-HW5/src/routers/groupRouter.ts

```diff
@@ -0,0 +1,73 @@
+import { Router, Response } from 'express';
+import { groupSchema, updateGroupSchema } from '../config/schemas';
+import { validateSchema } from '../utils/helpers';
+import { AppErrors } from '../logger/app-errors';
+import {
+  createGroup,
+  findGroupById,
+  getAllGroups,
+  updateGroup,
+  deleteGroupById,
+} from '../services/groupService';
+
+const router = Router();
+
+router.param('id', async (_, res: Response, next: () => void, id) => {
+  const group = await findGroupById(id);
+  if (!group) {
+    const errorMessage = `Group with id ${id} not found!`;
+    res.status(400).json(errorMessage);
+    throw new AppErrors(400, errorMessage, 'GET');
```

Comment:
Method name you're passing is ambiguous. Is it getting a user or a group or what?. Also, you need to include method args in the error.

Reply:
I don't understand why you need line 19, whereas you already have error handler that should be used when you throw AppError